File: rust/3d_ex/main.rs
use std::io::stdin;

fn cmp(a: i32, b: i32) -> i32 {
    if a == b {
        1
    } else {
        0
    }
}

fn init() -> i32 {
    let mut input = String::new();
    stdin().read_line(&mut input).expect("Failed to read line");

    let num = input.trim().parse().expect("Invalid input");

    num
}

fn main() {
    let n = init();
    if n <= 0 {
        panic!("Invalid input");
    }

    let mut a = init();
    let mut b = a;
    let mut count = 0;

    for _ in 1..n {
        a = init();
        count += cmp(a, b);
        b = a;
    }

    println!("{}", count);
}

File: c++/3d_ex/main.cpp
#include <iostream>

using namespace std;

int init(int val);
void err(string promt);
int counting_dup(int n);

int main() {
  int n;
  n = init(n);

  if (n <= 0)
    err("Invalid input");

  cout << counting_dup(n) << endl;

  return 0;
}

int counting_dup(int n) {
  int a = init(a), count = 0;
  int b = a;

  for (int i = 0; i < n - 1; i++) {
    a = init(a);
    if (a == b)
      count++;
    b = a;
  }
  return count;
}

void err(string promt) {
  cerr << promt << endl;
  exit(EXIT_FAILURE);
}

int init(int val) {
  if (!(cin >> val))
    err("Invalid input");
  return val;
}
